ml_model/model/feature_selection/feature_selection.py

```diff
@@ -0,0 +1,114 @@
+categorical_target = ["PBU", "BD50", "ABU50"]
+X = df[categorical_features]
+y = df[categorical_target]
+
+stat_header = {'input feature':'', 'output_feature': '', 'tau': 0, 'p-value':0}
+
+
+######################################
+## Classification feature selection ##
+######################################
+## Numeric input, categorical output
+
+## FILTER METHODS ##
+## write header
+with open('ml_model/src/main/model/feature_selection/kendall_features.csv', "w") as csv_file:
+    writer = csv.DictWriter(csv_file, fieldnames=stat_header.keys())
+    writer.writeheader()
+## Kendals Tau
+for feature in categorical_features:
+    for target in categorical_target:
+        tau,p = kendalltau(X[feature], y[target])
+        # write to csv
+        with open('ml_model/src/main/model/feature_selection/kendall_features.csv', "a") as csv_file:
+            writer = csv.DictWriter(csv_file, fieldnames=stat_header.keys())
+            writer.writerow({'input feature':feature, 'output_feature': target, 'tau': tau, 'p-value':p})
+## interprere the results
+# if p-value < 0.05, then the correlation is statistically significant. means that there is a relationship between the two variables.
+# filter based on above criteria and sort tau in descending order. now we have the most important features to predict the target variable.
```

Comment:
Fix: if |fau| > 0.1 and p-value < 0.05 then keep the feature

Reply:
We now don't compare features and targets to get the correlation. So this is outdated